front/views/settings/PreferencesView.tsx

```diff
@@ -1,89 +1,157 @@
-import React from 'react';
-import { View } from 'react-native';
-import { useDispatch, useSelector } from 'react-redux';
-import { Center, Button, Text, Switch, Slider, Select, Heading } from "native-base";
+import React from "react";
+import { View } from "react-native";
+import { useDispatch } from "react-redux";
+import {
+	Center,
+	Button,
+	Text,
+	Switch,
+	Slider,
+	Select,
+	Heading,
+} from "native-base";
 import { useLanguage } from "../../state/LanguageSlice";
-import i18n, { AvailableLanguages, DefaultLanguage, translate } from "../../i18n/i18n";
+import i18n, {
+	AvailableLanguages,
+	DefaultLanguage,
+	translate,
+	Translate,
+} from "../../i18n/i18n";
+import { RootState, useSelector } from "../../state/Store";
+import { SettingsState, updateSettings } from "../../state/SettingsSlice";
+import ElementList from "../../components/GtkUI/ElementList";
 
-const PreferencesView = ({navigation}) => {
-    const dispatch = useDispatch();
-    const language: AvailableLanguages = useSelector((state) => state.language.value);
+const PreferencesView = ({ navigation }) => {
+	const dispatch = useDispatch();
+	const language: AvailableLanguages = useSelector(
```

Comment:
Same as the comment regarding `useSelector`

Reply:
There's no useState here